### PR TITLE
MM-24597  Migrate API handler deleteChannel to be compatible with local mode

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Added local mode support
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24597

### CODE REVIEW

File: api4/channel_test.go [hunk 1]
```diff
@@ -1247,37 +1248,38 @@ func TestDeleteChannel(t *testing.T) {
...
-	require.NotNil(t, resp, "expected response to not be nil")
+		ch, err := th.App.GetChannel(publicChannel1.Id)
+		require.True(t, err != nil || ch.DeleteAt != 0, "should have failed to get deleted channel, or returned one with a populated DeleteAt.")
 
-	// successful delete of private channel
-	privateChannel2 := th.CreatePrivateChannel()
-	_, resp = Client.DeleteChannel(privateChannel2.Id)
-	CheckNoError(t, resp)
+		post1 := &model.Post{ChannelId: publicChannel1.Id, Message: "a" + GenerateTestId() + "a"}
+		_, resp = th.Client.CreatePost(post1)
```
Comment (mgdelacroix) [inline api4/channel_test.go:1262]: This should be the client injected by the helper:

```suggestion
		_, resp = client.CreatePost(post1)
```

File: api4/channel_test.go [hunk 2]
```diff
@@ -1247,37 +1248,38 @@ func TestDeleteChannel(t *testing.T) {
...
+		_, resp = client.DeleteChannel(publicChannel3.Id)
+		CheckNoError(t, resp)
 
+		// default channel cannot be deleted.
+		defaultChannel, _ := th.App.GetChannelByName(model.DEFAULT_CHANNEL, team.Id, false)
+		pass, resp = client.DeleteChannel(defaultChannel.Id)
+		CheckBadRequestStatus(t, resp)
+		require.False(t, pass, "should have failed")
+	})
 	// check system admin can delete a channel without any appropriate team or channel membership.
```
Comment (mgdelacroix) [inline api4/channel_test.go:1283]: Shouldn't these sysadmin tests be inside a `TestForSysadminAndLocal` helper?

File: api4/channel_test.go [hunk 3]
```diff
@@ -1241,80 +1242,81 @@ func TestSearchGroupChannels(t *testing.T) {
...
+	_, resp = c.DeleteChannel("junk")
 	CheckUnauthorizedStatus(t, resp)
 
-	Client.Logout()
-	_, resp = Client.DeleteChannel(GenerateTestId())
+	c.Logout()
+	_, resp = c.DeleteChannel(GenerateTestId())
 	CheckUnauthorizedStatus(t, resp)
 
 	_, resp = th.SystemAdminClient.DeleteChannel(publicChannel5.Id)
```
Comment (isacikgoz) [inline api4/channel_test.go:1322]: Maybe we can add `th.TestForSystemAdminAndLocal` here.

**Pull request merged**